backend/src/routes/user.ts

```diff
@@ -0,0 +1,179 @@
+import { Router, Request, Response } from 'express';
+import { ExerciseEvent, IExerciseEvent } from '../models/exercise-event';
+import { Types } from 'mongoose';
+import { User, IUser} from '../models/user';
+import { handleErrors } from '../util';
+
+async function createUser(req: Request, res: Response) {
+    let user: IUser = null;
+
+    try {
+        user = new User({
+            ...req.body
+        });
+    } catch (err) {
+        res.status(400).send({ message: 'Invalid user' });
+        return;
+    }
+
+    res.send(await user.save());
+}
+
+async function getAllUsers(req: Request, res: Response): Promise<void> {
+    res.send(await User.find({}));
+}
+
+async function getById(req: Request, res: Response) {
+    if (!Types.ObjectId.isValid(req.params.id)) {
+        res.status(404).send({ message: `User with id: ${req.params.id} not found` });
+        return;
+    }
+
+    const result = await User.findById(req.params.id);
+
+    if (!result) {
+        res.status(404).send({ message: `User with id: ${req.params.id} not found` });
+        return;
+    }
+
+    result.exerciseEvents.map(async(elem) => {
+        return await ExerciseEvent.findById(elem);
+    });
+
+    res.send({
+        _id: result._id,
+        username: result.username,
+        profile_image: result.profile_image,
+        bio: result.bio,
+        images: result.images,
+        exerciseEvents: result.exerciseEvents,
+    });
+}
+
+async function loginUser(req: Request, res: Response) {
+    let username = null;
+    let result = null;
+
+    try {
+        username = req.body.username;
+
+    } catch (err) {
+        console.error(err);
+        res.status(404).send({ message: `User with username: ${username} not found` });
+        return;
+    }
+
+    result = await User.findOne({ username: username });
+
+    if (!result) {
+        res.status(404).send({ message: `User with username: ${username} not found` });
+        return;
+    }
+
+    try {
+        let password = req.body.password;
```

Comment:
this should be `const`. how did the linter pass?

Reply:
you tell me, i didn't write it

Reply:
why is it in this PR then?